src/core/server/saved_objects/migrationsv2/model/model.ts

```diff
@@ -318,31 +318,42 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
     }
   } else if (stateP.controlState === 'CHECK_UNKNOWN_DOCUMENTS') {
     const res = resW as ExcludeRetryableEsError<ResponseType<typeof stateP.controlState>>;
+
+    const source = stateP.sourceIndex;
+    const target = stateP.versionIndex;
+    const nextState: State = {
+      ...stateP,
+      controlState: 'SET_SOURCE_WRITE_BLOCK',
+      sourceIndex: source,
+      targetIndex: target,
+      targetIndexMappings: disableUnknownTypeMappingFields(
+        stateP.targetIndexMappings,
+        stateP.sourceIndexMappings
+      ),
+      versionIndexReadyActions: Option.some<AliasAction[]>([
+        { remove: { index: source.value, alias: stateP.currentAlias, must_exist: true } },
+        { add: { index: target, alias: stateP.currentAlias } },
+        { add: { index: target, alias: stateP.versionAlias } },
+        { remove_index: { index: stateP.tempIndex } },
+      ]),
+    };
+
     if (Either.isRight(res)) {
-      const source = stateP.sourceIndex;
-      const target = stateP.versionIndex;
-      return {
-        ...stateP,
-        controlState: 'SET_SOURCE_WRITE_BLOCK',
-        sourceIndex: source,
-        targetIndex: target,
-        targetIndexMappings: disableUnknownTypeMappingFields(
-          stateP.targetIndexMappings,
-          stateP.sourceIndexMappings
-        ),
-        versionIndexReadyActions: Option.some<AliasAction[]>([
-          { remove: { index: source.value, alias: stateP.currentAlias, must_exist: true } },
-          { add: { index: target, alias: stateP.currentAlias } },
-          { add: { index: target, alias: stateP.versionAlias } },
-          { remove_index: { index: stateP.tempIndex } },
-        ]),
-      };
+      return nextState;
     } else {
       if (isLeftTypeof(res.left, 'unknown_docs_found')) {
         return {
-          ...stateP,
-          controlState: 'FATAL',
-          reason: extractUnknownDocFailureReason(res.left.unknownDocs, stateP.sourceIndex.value),
+          ...nextState,
```

Comment:
It's debatable whether or not this should be a left or a right result. I opted to keep it left since we just going to change this again to a failure case in the `master` branch for 8.0

Reply:
Correct approach is probably `right`. we don't need a `isRightTypeof`, as in `right` case, we would just returns the `unknownDocs` array in the `right` return of `checkForUnknownDocs` (and just append the log if `unknownDocs.length>0`). But that's not that important.

Reply:
FWIW, when I was working with Rudolf I understood that `left` results were those that ultimately led to an error state (i.e. migrations would fail). `right` results are all those that carry on with the 'happy path' through the migration algorithm. That being said, I also got the feeling were were trying to avoid error handling in the control flow which is kinda what we end up doing in a few cases.